src/main/assets/js/courts.ts

```diff
@@ -0,0 +1,65 @@
+import $ from 'jquery';
+import {orderToggleState} from '../../enums/searchToggleState';
+import {CourtsTableSearch} from './courts-table-search';
+
+export class CourtsController {
+
+  private contentId = '#main-content';
+  private tableContainerId = '#tableContainer';
+  private toggleClosedCourtsDisplay = 'toggleClosedCourtsDisplay';
+  private searchCourtsFilter = 'searchCourts';
+  private courtsNameAscToggleId = '#courtsNameAscToggle';
+  private courtsUpdatedAscToggleId = '#courtsUpdatedAscToggle';
+  private tableCourtsNameId = '#tableCourtsName';
+  private tableCourtsUpdatedId = '#tableCourtsUpdated';
+
+  constructor() {
+    this.initialize();
+  }
+
+  private initialize(): void {
+    $(() => {
+      if($(this.tableContainerId).length) { // so that this doesn't load on each page
+        CourtsTableSearch.setUpTable('name', true);
+        this.setUpToggleClosedCourtsDisplay();
+        this.setUpCourtsDynamicSearchFilter();
+        this.setUpAscDecNameFilter();
+        this.setUpAscDecUpdatedDateFilter();
+      }
+    });
+  }
+
+  private setUpToggleClosedCourtsDisplay(): void {
+    $(this.contentId).on('change', `input[name=${this.toggleClosedCourtsDisplay}]`, e => {
+      e.preventDefault();
+      $(this.courtsUpdatedAscToggleId).val() == orderToggleState.INACTIVE ?
+        CourtsTableSearch.setUpTable('name') : CourtsTableSearch.setUpTable('date');
+    });
+  }
+
+  private setUpCourtsDynamicSearchFilter(): void {
+    $(this.contentId).on('input', `input[name=${this.searchCourtsFilter}]`, e => {
+      e.preventDefault();
+      const updatedToggleValue = $(this.courtsUpdatedAscToggleId).val() as string;
+      (updatedToggleValue == orderToggleState.ASC || updatedToggleValue == orderToggleState.DESC) ?
+        CourtsTableSearch.setUpTable('date') : CourtsTableSearch.setUpTable('name');
+    });
+  }
+
+  private setUpAscDecNameFilter(): void {
+    $(this.contentId).on('click', `${this.tableCourtsNameId}`, e => {
+      e.preventDefault();
+      const ascToggle = $(this.courtsNameAscToggleId);
+      CourtsTableSearch.switchTableToggle(ascToggle, $(this.courtsUpdatedAscToggleId));
+      CourtsTableSearch.setUpTable('name');
+    });
+  }
+
+  private setUpAscDecUpdatedDateFilter(): void {
+    $(this.contentId).on('click', `${this.tableCourtsUpdatedId}`, e => {
```

Comment:
I pulled down the code and ordering by date column doesn't seem to be working correctly for me (I'm just clicking the updated column heading without any filters being applied). It does re-order the table but not entirely in date order.